app/assets/javascripts/models/data/spatial_condition.js.coffee

```diff
@@ -0,0 +1,72 @@
+ns = @edsc.models.data
+
+ns.SpatialCondition = do (ko, KnockoutModel=@edsc.models.KnockoutModel) ->
+  class SpatialCondition extends KnockoutModel
+    constructor: (spatial)->
+      @coordinates = ko.observable('')
+      @swPoint = ko.observable('')
+      @nePoint = ko.observable('')
+      @spatialType = ko.observable('')
+      @inputWidth = ko.observable(0)
+      @previousSpatial = ''
+      @previousType = ''
+      @visible = ko.observable(spatial().length > 0)
+
+      @error = @computed =>
+        coordinates = @coordinates()?.trim()
+        if coordinates? && coordinates.length > 0
+          coords = coordinates.split(/:/)
+          for coord in coords
+            ranges = coord.split(',')
+            return "Coordinate must be two comma-separated numbers: #{coord}" unless ranges.length == 2
+            for range in ranges
+              match = range.trim().match(/^-?\d+(\.\d+)?$/)
+              unless match?
+                return "Invalid coordinate: #{range}"
+              [all, min, max] = match
+              min = parseInt(min, 10)
+              max = parseInt(max, 10)
+        null
+
+      @queryCoordinates = @computed =>
+        @visible(spatial().length > 0)
+        type = spatial().split(':')[0]
+        newSpatial = ''
+
+        return null unless !!@previousType || !!spatial()
+
+        if type == @previousType
```

Comment:
I don't quite understand the keeping and testing against previousType and previousSpatial. This is a computed property; you should be able to just re-computed it without regard for its current value.